### PR TITLE
Simplify mail config

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

I was working on #6529 and started simplifying the code. I stopped when I realised that it's a bigger project and we need changes in ofn-install and secrets management. This commit is still useful though. It's just simplifying some legacy Spree code to help future developers understand it quicker.

#### What should we test?

- Choose a staging server with a working email setup.
- Emails should still work after deployment.
- Visit `/admin/mail_methods/edit` and untick email delivery.
- Verify that emails are not sent.
- Activate emails again.

#### Release notes

Changelog Category: Technical changes

Simplified email configuration code.

### CODE REVIEW

File: app/views/spree/admin/shared/_configuration_menu.html.haml [hunk 1]
```diff
@@ -5,8 +5,7 @@
   %nav.menu
     %ul.sidebar
       = configurations_sidebar_menu_item Spree.t(:general_settings), edit_admin_general_settings_path
-      - if Spree::Config[:override_actionmailer_config]
-        = configurations_sidebar_menu_item Spree.t(:mail_method_settings), edit_admin_mail_methods_path
+      = configurations_sidebar_menu_item Spree.t(:mail_method_settings), edit_admin_mail_methods_path
```
[the remark this answers is not in the document]
Comment (mkllnk) [reply app/views/spree/admin/shared/_configuration_menu.html.haml:8]: Not my fault. I made it shorter. :wink:

**Review by sauloperez — APPROVED** [review]

**Review by Matt-Yorkley — APPROVED** [review]
Nice :smile:

**Comment by mkllnk** [comment]
@Matt-Yorkley The failing specs revealed that I forgot another unused line of code. Would you mind double-checking my last commit?

**Review by sauloperez — APPROVED** [review]

**Comment by sauloperez** [comment]
This has been around quite a while and IMO it's fairly straightforward so I moved it to Test Ready.

**Comment by filipefurtad0** [comment]
Hey @mkllnk ,

Thank you for the detailed testing notes. The only thing not working was **disabling** of emails. After staging this PR on a staging server with a working email setup: 

i) Emails were working fine after deployment - tested several, but not all (order confirmation, order shipment, email with attachment invoice).
ii) Visit /admin/mail_methods/edit and untick email delivery - OK
iii) Verify that emails are not sent - here's the thing: emails got delivered anyway. But this was pre-existant to this PR, I checked other staging servers as well. It seems this field is not working - this probably needs an issue, right?
iv) Activate emails again - as mentioned, all working.

I think this should be ready to go, although I could not really assess iv).

**Pull request merged**